File: solutions/ocaml/rna-transcription/1/rna_transcription.ml
type dna = [ `A | `C | `G | `T ]
type rna = [ `A | `C | `G | `U ]

let f (x : dna) : rna =
  match x with
    | `G -> `C
    | `C -> `G
    | `T -> `A
    | `A -> `U

let to_rna (d : dna list) : rna list = List.map f d

File: solutions/zig/rna-transcription/1/rna_transcription.zig
const std = @import("std");
const mem = std.mem;

pub fn toRna(allocator: mem.Allocator, dna: []const u8) mem.Allocator.Error![]const u8 {
    var rna = try allocator.alloc(u8, dna.len);
    for (0..dna.len) |i| {
        rna[i] = switch (dna[i]) {
            'G' => 'C',
            'C' => 'G',
            'T' => 'A',
            'A' => 'U',
            else => dna[i],
        };
    }
    return rna;
}
